Source/WebCore/loader/ImageLoader.cpp

```diff
@@ -677,4 +678,27 @@ VisibleInViewportState ImageLoader::imageVisibleInViewport(const Document& docum
     return renderReplaced && renderReplaced->isContentLikelyVisibleInViewport() ? VisibleInViewportState::Yes : VisibleInViewportState::No;
 }
 
+bool ImageLoader::isAllowedToLoad(const ImageCandidate& candidate) const
+{
+#if ENABLE(WEB_ARCHIVE) || ENABLE(MHTML)
+    if (candidate.originAttribute == ImageCandidate::SrcOrigin)
+        return true;
+
+    Ref document = element().document();
+    RefPtr loader = document->loader();
+    if (!loader || !loader->hasArchiveResourceCollection())
```

Comment:
Just to confirm, this will ensure that the new logic only applies to web archives, and not loaded web pages right?

(which is necessary, since `<source>` selection in general is not based on an ability to load the content)

However, if this only applies to web archives, maybe the method name needs to be more specific?

Reply:
Yep, this logic is specific to web archives. I'll refactor this to make it more clear — e.g., `shouldIgnoreCandidateWhenLoadingFromArchive`